packages/amplify-category-auth/src/commands/auth/remove.ts

```diff
@@ -1,23 +1,30 @@
-export const name = 'remove';
-const category = 'auth';
-import { $TSContext, AmplifySupportedService, stateManager } from 'amplify-cli-core';
+import {
+  $TSContext, $TSMeta, AmplifyCategories, AmplifyError, AmplifySupportedService, stateManager,
+} from 'amplify-cli-core';
 import { printer } from 'amplify-prompts';
 import { messages } from '../../provider-utils/awscloudformation/assets/string-maps';
 import { AuthInputState } from '../../provider-utils/awscloudformation/auth-inputs-manager/auth-input-state';
 
-export const run = async (context: $TSContext) => {
+export const name = 'remove';
+const category = 'auth';
+
+/**
+ * Entry point for remove auth
+ */
+export const run = async (context: $TSContext): Promise<void> => {
   const { amplify, parameters } = context;
   const resourceName = parameters.first;
   const meta = stateManager.getMeta();
-  const dependentResources = Object.keys(meta).some(e => {
-    return ['analytics', 'api', 'storage', 'function'].includes(e) && Object.keys(meta[e]).length > 0;
-  });
-  if (dependentResources) {
-    printer.info(messages.dependenciesExists);
+
+  throwErrorIfProjectHasAnalytics(meta);
+
+  const hasPossiblyDependentResources = Object.keys(meta)
+    .some(categoryName => ['api', 'storage', 'function'].includes(categoryName) && Object.keys(meta[categoryName]).length > 0);
```

Comment:
why did analytics get removed from here? Does Analytics not depend on Auth?

Reply:
now that there's an upstream check to fail if analytics exists, we don't need to warn about it here

Reply:
I was about to ask a question here as well. Why do we have two mechanism for similar problem but they just differ by severity - i.e. one throws the other warns . (perhaps this should go to the long term follow up we should do here at some point).

Reply:
It's a good question that I don't know the answer to. I was just trying to keep the behavior change as localized as possible for this PR

Reply:
Ok, I see what's going on now; not a blocker IMO